spec/Section 3 -- Type System.md

```diff
@@ -1613,6 +1635,21 @@ is constructed with the following rules:
   does not provide a default value, the input object field definition's default
   value should be used.
 
+Further, if the input object is a OneOf Input Object, the following additional
+rules apply:
+
+- If the input object literal or unordered map does not contain exactly one
+  entry, an error must be thrown.
+
+- If the single entry in the input object literal or unordered map is {null}, an
+  error must be thrown.
```

Comment:
> or unordered map is {null}, an error must be thrown.

@benjie Does that mean that even if have a nullable arg of one input type, it is still required to pass value?

So if I have:
```graphql
input Options @oneOf {
  foo: String
  bar: Int
}

type Query {
  field(arg: Options): String
}
```
It basically means `arg` became required and following query is invalid:
```
{
  field
}
```

If so, I don't think it's right nullability should be controlled on the arg/input field and not on type as a whole.
Moreover, it changes the definition of required args/fields, previously it was "non-nullable arg/field without default value" now it became "non-nullable arg/field without default value or nullable field of oneOf input object type".

Reply:
Perhaps it means "if the single entry in the unordered map is {null}", rather than "if the unordered map is {null}", which makes more sense...?

Reply:
Ah; maybe that's ambiguous English? Perhaps you read it as:

> If (the single entry in the input object literal is {null}) or (unordered map is {null}), an error must be thrown.

whereas my intent was:

> If the single entry in (the input object literal or unordered map) is {null}, an error must be thrown.

i.e. if you supply a value for a oneof as either an input object literal (in the document) or an unordered map (i.e. via a variable), then the single entry in it must not be null. If you supply `null` as the argument value, then you are neither supplying an input object literal nor an unordered map.

For the avoidance of doubt, `{field}` is perfectly valid in the scenario you outline, and I agree it would be extremely undesirable were that not the case.

Reply:
Reworded :+1: